src/matvis/cli.py

```diff
@@ -264,6 +267,28 @@ def get_redundancies(bls, ndecimals: int = 2):
 
     return pairs
 
+# Chris: for now I have duplicated the get_redundancies function as a placeholder for generating the matrix sets and just assumed
+# 1x1 sub-matrices (the vector method).  In the future this will be replaced by a function that takes in a set of sub-matrices provided
+# by the user.
+def get_matrix_sets(bls, ndecimals: int = 2):
+    """Find redundant baselines."""
+    uvbins = set()
+    msets = []
+
+    # Everything here is in wavelengths
+    bls = np.round(bls, decimals=ndecimals)
+    nant = bls.shape[0]
+
+    # group redundant baselines
+    for i in range(nant):
+        for j in range(i + 1, nant):
+            u, v = bls[i, j]
+            if (u, v) not in uvbins and (-u, -v) not in uvbins:
+                uvbins.add((u, v))
+                msets.append([np.array([i]), np.array([j]))
+
+    return msets
```

Comment:
I think we can replace this with the little algorithm I had in my other branch. Also, put this in a new module (like "submatrices.py" or something)